### PR TITLE
Use `3` for Scala 3 Binary Version in Complete

### PR DESCRIPTION
**Pull request by LaurenceWarne**
Hi, I saw that that `3.x` is used by `Complete` when setting the binary version based on the full version for (non `-M`, `-RC`) Scala 3 versions, this PR changes it so `3` is always used to be in line with https://docs.scala-lang.org/scala3/reference/language-versions/binary-compatibility.html (and so completion results are a lot better when just using `withScalaVersion` :slightly_smiling_face:).

### CODE REVIEW

**Review by alexarchambault — COMMENTED** [review]
Thanks @LaurenceWarne, this looks good!

It would have been nice to add non regression tests for this in [`CompleteTests`](https://github.com/coursier/coursier/blob/1df62c141ba696a403ca34cbfbe871a318db639a/modules/coursier/shared/src/test/scala/coursier/complete/CompleteTests.scala). Adding tests to it requires updating some metadata under `modules/metadata` though. This isn't properly documented IIRC. The idea is to add tests, run them while ensuring [this variable](https://github.com/coursier/coursier/blob/1df62c141ba696a403ca34cbfbe871a318db639a/modules/coursier/jvm/src/test/scala/coursier/PlatformTestHelpers.scala#L38-L39) is true, then pushing the additional metadata of the `modules/metadata` git submodule to [this repository](https://github.com/coursier/test-metadata).

**Comment by LaurenceWarne** [comment]
Cool, I'll take a look!

**Comment by LaurenceWarne** [comment]
Is https://github.com/coursier/coursier/pull/2538/commits/de7192291d0eeed7b2dfe4c2336cd65be59274de along the lines of what you had in mind @alexarchambault :slightly_smiling_face: ?

File: .gitmodules [hunk 1]
```diff
@@ -1,6 +1,7 @@
 [submodule "tests/metadata"]
 	path = modules/tests/metadata
-	url = https://github.com/coursier/test-metadata.git
+	url = https://github.com/LaurenceWarne/test-metadata.git
+	branch = update-for-scala3-tests
```
Comment (LaurenceWarne) [inline .gitmodules:4]: Temporary until https://github.com/coursier/test-metadata/pull/6, hopefully tests should go green with it

**Comment by alexarchambault** [comment]
Merging, thanks!

**Pull request merged**